src/OpenTelemetry/Resources/OtelEnvResourceDetector.cs

```diff
@@ -22,17 +22,30 @@ namespace OpenTelemetry.Resources
 {
     internal class OtelEnvResourceDetector : IResourceDetector
     {
-        private const string OTelResourceEnvVarKey = "OTEL_RESOURCE_ATTRIBUTES";
+        public const string OTelResourceEnvVarKey = "OTEL_RESOURCE_ATTRIBUTES";
+
         private const char AttributeListSplitter = ',';
         private const char AttributeKeyValueSplitter = '=';
 
+        private readonly string environmentVariableName;
+
+        public OtelEnvResourceDetector(string environmentVariableName = null)
```

Comment:
I don't think we need to provide users to provide a custom env variable, as the spec is very clear that, if SDK choses to support ENV variable, then it must be the one specified by the spec.

Reply:
I can remove if you want. It's a pretty neat feature, though. Allows you to do stuff like:
```csharp
            var resource = new ResourceBuilder()
                .AddTelemetrySdk()
                .AddEnvironmentVariableDetector() // Default spec name
                .AddEnvironmentVariableDetector("MYCOMPANY_TELEMETRY") // Custom 1
                .AddEnvironmentVariableDetector("MYCOMPANY_PROD_TELEMETRY") // Custom 2
                .AddService("MyService")
                .AddAttributes(new Dictionary<string, object>
                {
                    ["service.datacenter"] = 1,
                })
                .Build();
```
LMK

Reply:
Don't hate me :D, but I'd vote for exposing minimal things for 1.0.0 release! if there is a need arising for this, then lets bring it through specs and to .NET.
Just sticking to the policy of "Additive changes to API is always possible, but breaking changes are not possible until major version bump" "_

Reply:
Removed